rest/main_legacy.go

```diff
@@ -43,6 +43,13 @@ func legacyServerMain(osArgs []string, flagStartupConfig *StartupConfig) error {
 		return err
 	}
 
+	initialStartupConfig, err := getInitialStartupConfig(migratedStartupConfig, flagStartupConfig)
+	if err != nil {
+		return err
+	}
+
+	ctx.initialStartupConfig = initialStartupConfig
```

Comment:
Does the same deep copy need to happen here to avoid the changing values inside pointers in startup config when running in non-persistent mode? And if so, should it be done inside `getInitialStartupConfig`?

Reply:
Yup. Sorry managed to miss that...
Good shout pushing it down